test/unit/server/handlers/consents/{ID}.test.ts

```diff
@@ -0,0 +1,325 @@
+
+/* Mock the ConsentCredential Value. */
+const credentialVerified: ConsentCredential = {
+  credentialType: 'FIDO',
+  credentialId: requestCredentialId,
+  credentialStatus: CredentialStatusEnum.VERIFIED,
+  credentialPayload: publicKey,
+  credentialChallenge: challenge
+}
+
+const externalScopes: ExternalScope[] = [
+  {
+    accountId: 'as2342',
+    actions: ['account.getAccess', 'account.transferMoney']
+  },
+  {
+    accountId: 'as22',
+    actions: ['account.getAccess']
+  }
+]
+
+// Mock Outgoing Request Body
+const requestBody: SDKStandardComponents.PutConsentsRequest = {
+  requestId: consentId,
+  scopes: externalScopes,
+  initiatorId: retrievedConsent.initiatorId as string,
+  participantId: retrievedConsent.participantId as string,
+  credential: {
+    id: requestCredentialId,
+    credentialType: 'FIDO',
+    status: CredentialStatusEnum.PENDING,
+    challenge: {
+      payload: retrievedConsent.credentialChallenge as string,
+      signature
+    },
+    payload: publicKey
+  }
+}
+
+describe('server/handler/consents/{ID}', (): void => {
+  beforeAll((): void => {
+    mockRetrieveValidConsent.mockResolvedValue(retrievedConsent)
+    mockPutConsents.mockResolvedValue(1 as unknown as SDKStandardComponents.GenericRequestResponse)
+    mockUpdateConsentCredential.mockResolvedValue(0)
+    mockBuildConsentRequestBody.mockResolvedValue(requestBody)
+    mockLoggerPush.mockReturnValue(null)
+    mockLoggerError.mockReturnValue(null)
+    mockVerifySignature.mockReturnValue(true)
+  })
+
+  beforeEach((): void => {
+    jest.clearAllMocks()
+  })
+
+  describe('validateAndUpdateConsent', (): void => {
+    it('should resolve successfully with no errors',
+      async (): Promise<void> => {
+        await expect(Handler.validateAndUpdateConsent(consentId, credentialRequest, destinationParticipantId))
+          .resolves
+          .toBeUndefined()
+
+        expect(mockRetrieveValidConsent).toHaveBeenCalledWith(consentId, challenge)
+
+        expect(mockVerifySignature).toHaveBeenCalledWith(challenge, signature, publicKey)
+        expect(mockUpdateConsentCredential).toHaveBeenCalledWith(retrievedConsent, credentialVerified)
+        expect(mockBuildConsentRequestBody).toHaveBeenCalledWith(retrievedConsent, signature, publicKey)
+        expect(mockPutConsents).toHaveBeenCalledWith(
+          consentId, requestBody, request.headers[Enum.Http.Headers.FSPIOP.SOURCE])
+
+        expect(mockLoggerError).not.toHaveBeenCalled()
+        expect(mockLoggerPush).not.toHaveBeenCalled()
+      })
+
+    it('should propagate retrieveValidConsent IncorrectChallenge error',
+      async (): Promise<void> => {
+        mockRetrieveValidConsent.mockRejectedValueOnce(new IncorrectChallengeError(consentId))
+
+        await expect(Handler.validateAndUpdateConsent(consentId, credentialRequest, destinationParticipantId)).resolves.toBeUndefined()
+
+        expect(mockLoggerPush).toBeCalledWith(new IncorrectChallengeError(consentId))
```

Comment:
I really don't like validating our tests based on what the logger did - perhaps we should think about how we are throwing and catching errors to we can test this properly.

Reply:
Ah i see - i think either works - as both are place holders for eventually making a `PUT .../error` call